### ISSUE TITLE
GraphQL add-on relies on deprecated Nashorn engine

### ISSUE THREAD

**Issue by globeone** [zaproxy/zaproxy#6501]
**Describe the bug**
Oracle Nashorn has been deprecated and deleted from ZAP but the ExtensionGraphQL 0.3.0 is still trying to access the library when ZAP boots.

**To Reproduce**
1. Go to 'zap.bat'
2. DoubleClick on 'zap.bat'
3. Scroll up to 'Failed to initialise extension org.zaproxy.addon.graphql.ExtensionGraphQl'

**Expected behavior**
The GraphQL extension to load without throwing errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/10052587/112826807-c5e77180-908d-11eb-96ac-c8cb8f5bf2b7.png)

**Software versions**
 - ZAP: D-2021-03-22
 - Add-on: GraphQL 0.3.0
 - OS: Windows 10
 - Java: 
 openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36)
OpenJDK 64-Bit Server VM (build 15+36, mixed mode, sharing)
 - Browser: Google Chrome Version 89.0.4389.90 (Official Build) (64-bit)

**Errors from the zap.log file**
```
15879 [ZAP-BootstrapGUI] ERROR org.parosproxy.paros.extension.ExtensionLoader - Failed to initialise extension org.zaproxy.addon.graphql.ExtensionGraphQl (from add-on [id=graphql, version=0.3.0]), cause: InvalidParameterException: No such engine: Oracle Nashorn
java.security.InvalidParameterException: No such engine: Oracle Nashorn
        at org.zaproxy.zap.extension.script.ExtensionScript.getEngineWrapper(ExtensionScript.java:472) ~[zap-D-2021-03-22.jar:D-2021-03-22]
        at org.zaproxy.addon.graphql.ExtensionGraphQl.addScript(ExtensionGraphQl.java:192) ~[?:?]
        at org.zaproxy.addon.graphql.ExtensionGraphQl.postInit(ExtensionGraphQl.java:109) ~[?:?]
        at org.parosproxy.paros.extension.ExtensionLoader.hookAllExtension(ExtensionLoader.java:934) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.parosproxy.paros.extension.ExtensionLoader.startLifeCycle(ExtensionLoader.java:798) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.parosproxy.paros.control.AbstractControl.loadExtension(AbstractControl.java:58) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.parosproxy.paros.control.Control.init(Control.java:141) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.parosproxy.paros.control.Control.initSingletonWithView(Control.java:361) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.zaproxy.zap.GuiBootstrap.initControlAndPostViewInit(GuiBootstrap.java:221) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.zaproxy.zap.GuiBootstrap.access$100(GuiBootstrap.java:64) [zap-D-2021-03-22.jar:D-2021-03-22]
        at org.zaproxy.zap.GuiBootstrap$2.run(GuiBootstrap.java:175) [zap-D-2021-03-22.jar:D-2021-03-22]
        at java.lang.Thread.run(Thread.java:832) [?:?]
```

**Additional context**
Straightforward dependency issue after deprecation.

**Would you like to help fix this issue?**
Happy to test.

**Comment by thc202** [comment]
The latest version available in the marketplace no longer fails at the start.

The add-on still needs to be changed to use other script engine going forward.

**Comment by thc202** [comment]
Addressed in zaproxy/zap-extensions#4244.
